modules/git/commit.go

```diff
@@ -375,32 +374,13 @@ func (c *Commit) GetSubModules() (*ObjectCache, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	defer rd.Close()
-	scanner := bufio.NewScanner(rd)
-	c.submoduleCache = newObjectCache()
-	var ismodule bool
-	var path string
-	for scanner.Scan() {
-		if strings.HasPrefix(scanner.Text(), "[submodule") {
-			ismodule = true
-			continue
-		}
-		if ismodule {
-			fields := strings.Split(scanner.Text(), "=")
-			k := strings.TrimSpace(fields[0])
-			if k == "path" {
-				path = strings.TrimSpace(fields[1])
-			} else if k == "url" {
-				c.submoduleCache.Set(path, &SubModule{path, strings.TrimSpace(fields[1])})
-				ismodule = false
-			}
-		}
-	}
-	if err = scanner.Err(); err != nil {
-		return nil, fmt.Errorf("GetSubModules scan: %w", err)
-	}
 
+	r := io.LimitReader(rd, 100*1024) // limit to 100KB
```

Comment:
We use submodules in funny way (to represent packages in a Linux distribution :smile: ) and our file is 250kB (5k submodules). 1MB for openSUSE Tumbleweed is possible here. Larger if not using relative URLs

Is there really a need to impose such small limit like this here? Previously there wasn't a limit.

Reply:
This should read "2MB is possible for openSUSE Tumbleweed" :laughing:

Reply:
let's remove the limit and keep the old logic.